security/subsystem/src/main/java/org/jboss/as/security/SecurityDomainReloadRemoveHandler.java

```diff
@@ -38,6 +38,14 @@ public SecurityDomainReloadRemoveHandler() {
     }
 
     @Override
+    protected void updateModel(OperationContext context, ModelNode operation) throws OperationFailedException {
+        // verify that the resource exist before removing it
+        context.readResource(PathAddress.EMPTY_ADDRESS, false);
+
+        super.updateModel(context, operation);
+    }
```

Comment:
This is an good fix for the EAP 6 and 7 CP branches, as it's conservative by being limited to this one case. But the upstream fix should be in the superclass in core so it will apply to all cases.

The JIRA should be moved to WFCORE.